lib/util/move-file.js

```diff
@@ -25,9 +25,11 @@ function moveFile (src, dest) {
       if (err) {
         if (err.code === 'EEXIST' || err.code === 'EBUSY') {
           // file already exists, so whatever
-        } else if (err.code === 'EPERM' && process.platform === 'win32') {
-          // file handle stayed open even past graceful-fs limits
         } else {
+          /* istanbul ignore next */
+          if (err.code === 'EPERM' && process.platform === 'win32') {
+            // file handle stayed open even past graceful-fs limits
+          }
```

Comment:
@isaacs I couldn't get the `/* istanbul ignore if */` to work as the [example you set up](https://gist.github.com/isaacs/987fc367e2422ba782de9298586ea5c7#file-ignore-if-works-js) 😵  I wonder if it's the version of `nyc` we're running or if I'm messing up with something 🤔 

it did worked now with the `/* istanbul ignore next */` and in my understanding that's fine now since there's no `else` as a part of that same block

Reply:
So, just a pedantic note, these are not the same thing:

```js
/* istanbul ignore next */
if (conditionNeverMet) {
  foo()
}

/* istanbul ignore if */
if (conditionNeverMet) {
  foo()
}
```

The first ignores both the if _and_ else paths.  The second ignores only the `if` branch, but verifies that the `else` branch was hit.

They appear the same in this case because the `else` path _is_ getting hit.  But if at some point that changed, and instead of never being satisfied, it was _always_ being satisfied, then the `/* istanbul ignore if */` would show missing branch coverage, and `/* istanbul ignore next */` would not.

In other words, from istanbul's point of view _every_ `if` has an `else`, even if there's no functional code there.  Imagine that it sees the code like this:

```js
if (condition) {
  doSomething()
} else {
  doNothing()
}
```

and then it's verifying that the `doNothing` branch is being traversed.

In general, it's best to ignore as little as possible.

Reply:
thanks for the note! that confirms my understanding of it 👍 

I can try to get down to the bottom of the problem later 😊